src/main/kotlin/cacophony/grammars/Analysis.kt

```diff
@@ -2,43 +2,45 @@ package cacophony.grammars
 
 import cacophony.automata.DFA
 
-typealias DFAStateReference<StateType, AtomType, ResultType> = Pair<StateType, DFA<StateType, AtomType, ResultType>>
+typealias DFAStateReference<StateType, SymbolType> = Pair<StateType, DFA<StateType, SymbolType, Production<SymbolType>>>
 
 fun <StateType, SymbolType> findNullable(
     automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
-): Collection<DFAStateReference<StateType, SymbolType, Production<SymbolType>>> {
+): Collection<DFAStateReference<StateType, SymbolType>> {
     TODO("Not implemented")
 }
 
 fun <StateType, SymbolType> findFirst(
     automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
-    nullable: Collection<DFAStateReference<StateType, SymbolType, Production<SymbolType>>>,
-): Map<DFAStateReference<StateType, SymbolType, Production<SymbolType>>, Collection<SymbolType>> {
+    nullable: Collection<DFAStateReference<StateType, SymbolType>>,
+): Map<DFAStateReference<StateType, SymbolType>, Collection<SymbolType>> {
     TODO("Not implemented")
 }
 
 fun <StateType, SymbolType> findFollow(
     automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
-    nullable: Collection<DFAStateReference<StateType, SymbolType, Production<SymbolType>>>,
-    first: Map<DFAStateReference<StateType, SymbolType, Production<SymbolType>>, Collection<SymbolType>>,
-): Map<DFAStateReference<StateType, SymbolType, Production<SymbolType>>, Collection<SymbolType>> {
+    nullable: Collection<DFAStateReference<StateType, SymbolType>>,
+    first: Map<DFAStateReference<StateType, SymbolType>, Collection<SymbolType>>,
+): Map<DFAStateReference<StateType, SymbolType>, Collection<SymbolType>> {
     TODO("Not implemented")
 }
 
 data class AnalyzedGrammar<StateType, SymbolType>(
     val syncSymbolTypes: Collection<SymbolType>,
     val automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
+    val nullable: Collection<DFAStateReference<StateType, SymbolType>>,
+    val first: Map<DFAStateReference<StateType, SymbolType>, Collection<SymbolType>>,
+    val follow: Map<DFAStateReference<StateType, SymbolType>, Collection<SymbolType>>,
```

Comment:
imo it's better to have those sets here, so we can test processing `AnalyzedGrammar` without having `find....` implemented yet